src/mds/Server.cc

```diff
@@ -8927,14 +8953,15 @@ void Server::handle_client_renamesnap(MDRequestRef& mdr)
   dout(10) << " stid is " << stid << dendl;
 
   // journal
-  inode_t *pi = diri->project_inode();
-  pi->ctime = mdr->get_op_stamp();
-  pi->version = diri->pre_dirty();
+  auto &pi = diri->project_inode(false, true);
+  pi.inode.ctime = mdr->get_op_stamp();
+  pi.inode.version = diri->pre_dirty();
 
   // project the snaprealm
-  sr_t *newsnap = diri->project_snaprealm();
-  assert(newsnap->snaps.count(snapid));
-  newsnap->snaps[snapid].name = dstname;
+  auto &newsnap = *pi.snapnode;
+  auto em = newsnap.snaps.emplace(std::piecewise_construct, std::forward_as_tuple(snapid), std::forward_as_tuple());
+  assert(em.second); /* actually inserted */
+  em.first->second.name = dstname;
```

Comment:
diff --git a/src/mds/Server.cc b/src/mds/Server.cc
index 0d0706c812..b58694945a 100644
--- a/src/mds/Server.cc
+++ b/src/mds/Server.cc
@@ -9367,9 +9367,9 @@ void Server::handle_client_renamesnap(MDRequestRef& mdr)
 
   // project the snaprealm
   auto &newsnap = *pi.snapnode;
-  auto em = newsnap.snaps.emplace(std::piecewise_construct, std::forward_as_tuple(snapid), std::forward_as_tuple());
-  assert(em.second); /* actually inserted */
-  em.first->second.name = dstname;
+  auto it = newsnap.snaps.find(snapid);
+  assert(it != newsnap.snaps.end());
+  it->second.name = dstname;
 
   // journal the inode changes
   mdr->ls = mdlog->get_current_segment();